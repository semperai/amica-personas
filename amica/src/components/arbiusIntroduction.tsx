import { useAccount } from 'wagmi'
import { ConnectButton } from '@rainbow-me/rainbowkit';
import { useState } from "react";
import { useTranslation } from 'react-i18next';

export const ArbiusIntroduction = ({ open, close }: {
  open: boolean;
  close: () => void;
}) => {
  const { t } = useTranslation();
  const { address } = useAccount();


  if (! open) {
    return null;
  }

  return (
    <div className="absolute z-40 h-full w-full bg-black/30 mx-auto sm:px-24 lg:px-32 font-M_PLUS_2"
    >
      <div className="mx-auto max-h-full overflow-auto rounded-lg bg-white/80 p-4 backdrop-blur-lg shadow-lg">
        <div className="my-4">
          <div className="my-8 font-bold text-xl">
            {t("Welcome to Amica by Arbius")}
          </div>
          <p>{t("arbius_intro", `
            Amica is the best way for AI to interact with humans. The Arbius interface shows how people may interact with decentralized AI. With this setup, anyone may prompt Amica by spending a small amount of AIUS tokens, and when an Arbius miner solves the prompt, the users connected will receive a response. This is set up as a global demonstration, so do not expect any privacy.
          `)}
          </p>
        </div>

        <div className="my-8 flex space-x-4">
          <ConnectButton />
        </div>
        <div className="my-8">
          {address && (
            <button
              onClick={() => {
                close();
              }}
              className="ml-3 inline-flex items-center rounded-md bg-white px-2.5 py-1.5 text-sm font-semibold text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 hover:bg-gray-50"
            >
              {t("Continue to Amica")}
            </button>
          )}
          {(!address) && (
            <p>{t("Please connect your wallet to continue.")}</p>
          )}
        </div>
      </div>
    </div>
  );
};
