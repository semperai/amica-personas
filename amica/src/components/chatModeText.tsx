import { useEffect, useRef, useState } from "react";
import { clsx } from "clsx";
import { config } from "@/utils/config";
import { IconButton } from "./iconButton";
import { useTranslation } from "react-i18next";
import { Message } from "@/features/chat/messages";

export const ChatModeText = ({ messages }: { messages: Message[] }) => {
    const chatScrollRef = useRef<HTMLDivElement>(null);

    useEffect(() => {
        chatScrollRef.current?.scrollIntoView({
            behavior: "smooth",
            block: "center",
        });
    }, [messages]);

    return (
        <div className="fixed bottom-0 w-full h-[90%] mb-20 flex flex-col justify-end">
            <div className="w-full h-full overflow-y-auto flex flex-col-reverse">

                <div className="w-full max-w-full mx-auto px-4 md:px-16 flex flex-col">
                    {messages.map((msg, i) => {
                        return (
                            <div key={i} ref={messages.length - 1 === i ? chatScrollRef : null}>
                                <Chat
                                    role={msg.role}
                                    message={(msg.content as string).replace(/\[(.*?)\]/g, "")}
                                    num={i}
                                />
                            </div>
                        );
                    })}
                </div>
            </div>
        </div>

    );
};

function Chat({
    role,
    message,
    num,
}: {
    role: string;
    message: string;
    num: number;
}) {
    const { t } = useTranslation();
    const scrollRef = useRef<HTMLDivElement>(null);
    const [unlimited, setUnlimited] = useState(false);

    // useEffect(() => {
    //     scrollRef.current?.scrollIntoView({
    //         behavior: "smooth",
    //         block: "center",
    //     });
    // });

    return (
        <div className={clsx(
            'mx-auto max-w-4xl my-2',
            role === "assistant" ? "pr-10 sm:pr-20" : "pl-10 sm:pl-20",
        )}>
            <div className="backdrop-blur-lg rounded-lg">
                <div className="shadow-lg backdrop-blur-lg rounded-lg bg-white/70">

                    <div className={clsx(
                        'pr-1 py-3 bg-rose/90 rounded-t-lg text-white font-bold tracking-wider',
                        role === "assistant" ? "px-8" : "px-8",
                    )}>
                        <span className={clsx(
                            "p-4 rounded-lg rounded-tl-none rounded-tr-none shadow-sm",
                            role === "assistant" ? "bg-pink-600/80" : "bg-cyan-600/80",
                        )}>
                            {role === "assistant" && config('name').toUpperCase()}
                            {role === "user" && t("YOU")}
                        </span>

                        {role === "assistant" && (
                            <IconButton
                                iconName="24/FrameSize"
                                className="bg-transparent hover:bg-transparent active:bg-transparent disabled:bg-transparent float-right"
                                isProcessing={false}
                                onClick={() => setUnlimited(!unlimited)}
                            />

                        )}
                    </div>
                    {role === "assistant" && (
                        <div className={clsx(
                            "px-8 py-4 overflow-y-auto",
                            unlimited ? 'max-h-32' : 'max-h-[calc(75vh)]',
                        )}>
                            <div className="min-h-8 max-h-full typography-16 font-bold text-gray-600">
                                {message.replace(/\[([a-zA-Z]*?)\]/g, "")}
                                <div ref={scrollRef} />
                            </div>
                        </div>
                    )}
                    {role === "user" && (
                        <div className="px-8 py-4 max-h-32 overflow-y-auto">
                            <div className="min-h-8 max-h-full typography-16 font-bold text-gray-600">
                                {message.replace(/\[([a-zA-Z]*?)\]/g, "")}
                                <div ref={scrollRef} />
                            </div>
                        </div>
                    )}
                </div>
            </div>


        </div>

    );
}
