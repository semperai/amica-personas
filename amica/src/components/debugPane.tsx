import { useEffect, useRef, useState, useMemo } from "react";
import { Switch } from '@headlessui/react'
import { useKeyboardShortcut } from "@/hooks/useKeyboardShortcut";
import { clsx } from "clsx";
import { config } from "@/utils/config";

const TOTAL_ITEMS_TO_SHOW = 50;

// Safe JSON stringify that handles circular references
function safeStringify(obj: any): string {
  try {
    const seen = new WeakSet();
    return JSON.stringify(obj, (key, value) => {
      if (typeof value === 'object' && value !== null) {
        if (seen.has(value)) {
          return '[Circular]';
        }
        seen.add(value);
      }
      return value;
    }, 2);
  } catch (e) {
    return String(obj);
  }
}

function SwitchToggle({ enabled, set }: {
  enabled: boolean;
  set: (enabled: boolean) => void;
}) {
  return (
    <Switch
      className="group ml-1 relative inline-flex h-5 w-10 flex-shrink-0 cursor-pointer items-center justify-center rounded-full focus:outline-none focus:ring-0"
      checked={enabled}
      onChange={set}
    >
      <span className="sr-only">Use setting</span>
      <span aria-hidden="true" className="pointer-events-none absolute h-full w-full rounded-md" />
      <span
        aria-hidden="true"
        className={clsx(
          enabled ? 'bg-indigo-200' : 'bg-gray-200',
          'pointer-events-none absolute mx-auto h-4 w-9 rounded-full transition-colors duration-200 ease-in-out'
        )}
      />
      <span
        aria-hidden="true"
        className={clsx(
          enabled ? 'translate-x-5' : 'translate-x-0',
          'pointer-events-none absolute left-0 inline-block h-5 w-5 transform rounded-full border border-gray-200 bg-white shadow ring-0 transition-transform duration-200 ease-in-out'
        )}
        />
    </Switch>
  )
}

export function DebugPane({ onClickClose }: {
  onClickClose: () => void
}) {
  const [typeDebugEnabled, setTypeDebugEnabled] = useState(false);
  const [typeInfoEnabled, setTypeInfoEnabled] = useState(true);
  const [typeWarnEnabled, setTypeWarnEnabled] = useState(true);
  const [typeErrorEnabled, setTypeErrorEnabled] = useState(true);
  const [isReady, setIsReady] = useState(false);

  const scrollRef = useRef<HTMLDivElement>(null);
  const modalRef = useRef<HTMLDivElement>(null);

  useKeyboardShortcut("Escape", onClickClose);

  // Close on click outside
  useEffect(() => {
    const handleClickOutside = (event: MouseEvent) => {
      if (modalRef.current && !modalRef.current.contains(event.target as Node)) {
        onClickClose();
      }
    };

    document.addEventListener('mousedown', handleClickOutside);
    return () => document.removeEventListener('mousedown', handleClickOutside);
  }, [onClickClose]);

  // Get logs safely and defer processing
  const filteredLogs = useMemo(() => {
    // Don't process logs until modal is ready
    if (!isReady) return [];

    const logs = (window as any).error_handler_logs || [];
    const recentLogs = logs.slice(-TOTAL_ITEMS_TO_SHOW);

    return recentLogs.filter((log: any) => {
      if (log.type === 'debug' && !typeDebugEnabled) return false;
      if ((log.type === 'info' || log.type === 'log') && !typeInfoEnabled) return false;
      if (log.type === 'warn' && !typeWarnEnabled) return false;
      if (log.type === 'error' && !typeErrorEnabled) return false;
      return true;
    });
  }, [typeDebugEnabled, typeInfoEnabled, typeWarnEnabled, typeErrorEnabled, isReady]);

  // Defer heavy rendering until after modal opens
  useEffect(() => {
    // Use requestAnimationFrame for better timing
    const rafId = requestAnimationFrame(() => {
      setTimeout(() => {
        setIsReady(true);
        requestAnimationFrame(() => {
          scrollRef.current?.scrollIntoView({
            behavior: "auto",
            block: "center",
          });
        });
      }, 100);
    });

    return () => cancelAnimationFrame(rafId);
  }, []);

  function onClickCopy() {
    try {
      const logs = (window as any).error_handler_logs || [];
      navigator.clipboard.writeText(safeStringify(logs));
    } catch (e) {
      console.error('Failed to copy logs:', e);
    }
  }

  return (
    <div className="fixed inset-0 bg-black/60 backdrop-blur-md z-[100] flex items-center justify-center overflow-hidden p-4 md:p-8 animate-in fade-in duration-200">
      <div ref={modalRef} className="w-full h-full max-w-7xl bg-slate-50 shadow-2xl flex flex-col md:max-h-[90vh] md:rounded-lg overflow-hidden">
        {/* Header */}
        <div className="flex-shrink-0 bg-slate-900 text-white p-5 shadow-lg border-b border-slate-700/50">
          <div className="flex items-center justify-between">
            <div className="flex items-center gap-3">
              <div className="bg-white/20 backdrop-blur-sm rounded-lg p-2">
                <svg className="w-6 h-6" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                  <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M9 12h6m-6 4h6m2 5H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z" />
                </svg>
              </div>
              <div>
                <h2 className="text-xl font-bold tracking-tight">Debug Console</h2>
                <p className="text-xs text-white/70">System logs and diagnostics</p>
              </div>
              <button
                onClick={onClickCopy}
                className="ml-2 bg-white/10 hover:bg-white/20 active:bg-white/30 backdrop-blur-sm text-white rounded-lg px-3 py-2 transition-all hover:scale-105 cursor-pointer text-sm font-medium"
                title="Copy logs to clipboard"
              >
                Copy Logs
              </button>
            </div>
            <button
              onClick={onClickClose}
              className="bg-white/10 hover:bg-white/20 active:bg-white/30 backdrop-blur-sm text-white rounded-lg p-2.5 transition-colors cursor-pointer"
            >
              <svg className="w-5 h-5" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M6 18L18 6M6 6l12 12" />
              </svg>
            </button>
          </div>

          {/* System Info */}
          <div className="mt-4 flex flex-wrap gap-2 text-xs">
            <span className="bg-white/15 backdrop-blur-sm px-3 py-1.5 rounded-lg border border-white/20">
              <span className="font-semibold text-white/90">LLM:</span> <span className="text-white/70">{config("chatbot_backend")}</span>
            </span>
            <span className="bg-white/15 backdrop-blur-sm px-3 py-1.5 rounded-lg border border-white/20">
              <span className="font-semibold text-white/90">TTS:</span> <span className="text-white/70">{config("tts_backend")}</span>
            </span>
            <span className="bg-white/15 backdrop-blur-sm px-3 py-1.5 rounded-lg border border-white/20">
              <span className="font-semibold text-white/90">STT:</span> <span className="text-white/70">{config("stt_backend")}</span>
            </span>
            <span className="bg-white/15 backdrop-blur-sm px-3 py-1.5 rounded-lg border border-white/20">
              <span className="font-semibold text-white/90">Build:</span> <span className="text-white/70">{import.meta.env.VITE_CONFIG_BUILD_ID}</span>
            </span>
          </div>
        </div>

        {/* Filters */}
        <div className="flex-shrink-0 bg-slate-800 border-b border-slate-700 p-4 shadow-sm">
          <div className="flex flex-wrap gap-2.5">
            <span className="inline-flex items-center gap-2 rounded-lg bg-slate-700 px-4 py-2 text-sm font-semibold text-slate-200 shadow-sm border border-slate-600 hover:bg-slate-600 transition-colors">
              Debug
              <SwitchToggle enabled={typeDebugEnabled} set={setTypeDebugEnabled} />
            </span>
            <span className="inline-flex items-center gap-2 rounded-lg bg-emerald-900/40 px-4 py-2 text-sm font-semibold text-emerald-300 shadow-sm border border-emerald-700/50 hover:bg-emerald-900/60 transition-colors">
              Info
              <SwitchToggle enabled={typeInfoEnabled} set={setTypeInfoEnabled} />
            </span>
            <span className="inline-flex items-center gap-2 rounded-lg bg-amber-900/40 px-4 py-2 text-sm font-semibold text-amber-300 shadow-sm border border-amber-700/50 hover:bg-amber-900/60 transition-colors">
              Warning
              <SwitchToggle enabled={typeWarnEnabled} set={setTypeWarnEnabled} />
            </span>
            <span className="inline-flex items-center gap-2 rounded-lg bg-rose-900/40 px-4 py-2 text-sm font-semibold text-rose-300 shadow-sm border border-rose-700/50 hover:bg-rose-900/60 transition-colors">
              Error
              <SwitchToggle enabled={typeErrorEnabled} set={setTypeErrorEnabled} />
            </span>
          </div>
        </div>

        {/* Log Content */}
        <div className="flex-1 overflow-y-auto bg-slate-950 p-5">
          {!isReady ? (
            <div className="flex items-center justify-center h-full text-slate-400">
              <div className="text-center">
                <div className="animate-spin rounded-full h-10 w-10 border-3 border-slate-700 border-t-slate-400 mx-auto mb-3"></div>
                <div className="text-sm font-medium">Loading logs...</div>
              </div>
            </div>
          ) : (
            <div className="space-y-1.5 font-mono text-xs">
              {filteredLogs.map((log: any, idx: number) => {
                // Prepare log message string (lightweight)
                let logMessage = '';
                try {
                  logMessage = [...log.arguments].map((v: any) =>
                    typeof v === 'object' ? '[Object]' : String(v)
                  ).join(" ");
                  if (logMessage.length > 500) {
                    logMessage = logMessage.substring(0, 500) + '...';
                  }
                } catch (e) {
                  logMessage = '[Error rendering log]';
                }

                return (
                  <div
                    key={log.ts+idx}
                    className={clsx(
                      "rounded px-3 py-2 flex items-start gap-3 border transition-colors hover:bg-slate-900/50",
                      log.type === 'error' && 'bg-red-950/20 border-red-900/30',
                      log.type === 'warn' && 'bg-amber-950/10 border-amber-900/20',
                      log.type === 'debug' && 'bg-slate-900/30 border-slate-800/50',
                      (log.type === 'info' || log.type === 'log') && 'bg-emerald-950/10 border-emerald-900/20',
                    )}
                  >
                    {log.type === 'debug' && (
                      <span className="px-2 py-1 text-[10px] font-bold text-slate-400 bg-slate-800 rounded uppercase flex-shrink-0 border border-slate-700">
                        DBG
                      </span>
                    )}
                    {(log.type === 'info' || log.type === 'log') && (
                      <span className="px-2 py-1 text-[10px] font-bold text-emerald-400 bg-emerald-950/60 rounded uppercase flex-shrink-0 border border-emerald-800">
                        INF
                      </span>
                    )}
                    {log.type === 'warn' && (
                      <span className="px-2 py-1 text-[10px] font-bold text-amber-400 bg-amber-950/60 rounded uppercase flex-shrink-0 border border-amber-800">
                        WRN
                      </span>
                    )}
                    {log.type === 'error' && (
                      <span className="px-2 py-1 text-[10px] font-bold text-rose-400 bg-rose-950/60 rounded uppercase flex-shrink-0 border border-rose-800">
                        ERR
                      </span>
                    )}

                    <span className="text-slate-500 text-[10px] flex-shrink-0 font-semibold pt-1">
                      {new Date(log.ts).toLocaleTimeString()}
                    </span>

                    <span className="text-slate-200 flex-1 break-all leading-relaxed">
                      {logMessage}
                    </span>
                  </div>
                );
              })}
              <div ref={scrollRef} />
            </div>
          )}
        </div>
      </div>
    </div>
  );
}
