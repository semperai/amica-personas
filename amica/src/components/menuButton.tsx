import * as React from 'react';
import clsx from 'clsx';

export function MenuButton({
  icon,
  onClick,
  large,
  label,
  href,
  target,
  disabled,
}: {
  icon: React.ComponentType<any>,
  onClick?: () => void,
  large: boolean,
  label: string,
  href?: string,
  target?: string,
  disabled?: boolean,
}) {
  const Icon = icon; // Capitalize to use as component

  if (href) {
    onClick = () => {
      window.open(href, target);
    };
  }

  return (
    <div className="flex flex-row items-center space-x-2">
      <button
        disabled={disabled}
        onClick={onClick}
      >
        <Icon
          className={clsx(
            large ? 'h-14 w-14' : 'h-7 w-7',
            'text-white',
            disabled && 'cursor-not-allowed opacity-20',
            !disabled && 'opacity-50 hover:opacity-100 active:opacity-100 hover:cursor-pointer',
         )}
          aria-hidden="true"
        />
        <span className="text-white hidden">{label}</span>
      </button>
    </div>
  );
}
