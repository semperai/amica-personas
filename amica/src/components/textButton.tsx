import { ButtonHTMLAttributes } from "react";
type Props = ButtonHTMLAttributes<HTMLButtonElement>;

export const TextButton = (props: Props) => {
  return (
    <button
      {...props}
      className={`px-3 py-1.5 text-white text-xs font-semibold bg-slate-800 hover:bg-slate-700 active:bg-slate-900 disabled:bg-slate-400 rounded transition-all cursor-pointer ${props.className}`}
    >
      {props.children}
    </button>
  );
};
