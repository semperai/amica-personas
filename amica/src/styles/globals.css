@import "tailwindcss";

@theme {
  --color-primary: #856292;
  --color-primary-hover: #8E76A1;
  --color-primary-press: #988BB0;
  --color-primary-disabled: #6F48694D;
  --color-secondary: #FF617F;
  --color-secondary-hover: #FF849B;
  --color-secondary-press: #FF9EB1;
  --color-secondary-disabled: #FF617F4D;
  --color-base: #FBE2CA;
  --color-text-primary: #514062;

  --font-family-fredoka: Fredoka, sans-serif;
  --font-family-m-plus-2: Montserrat, M_PLUS_2, sans-serif;
  --font-family-montserrat: Montserrat, sans-serif;
}

body {
  background-position: top center;
  background-attachment: fixed;
  background-size: cover;
  background-repeat: no-repeat;
}

@utility scroll-hidden {
  -ms-overflow-style: none;
  scrollbar-width: none;

  &::-webkit-scrollbar {
    display: none;
  }
}
