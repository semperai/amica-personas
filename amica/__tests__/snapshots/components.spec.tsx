import { describe, expect, test } from "@jest/globals";
import { render } from "@testing-library/react";
import { IconButton } from "@/components/iconButton";
import { TextButton } from "@/components/textButton";

/**
 * Component Snapshot Tests
 *
 * These tests capture the rendered output of components and detect unintended changes.
 * If a snapshot fails, review the changes and update with `npm test -- -u` if intended.
 */

describe("Button Components", () => {
  describe("IconButton", () => {
    test("should match snapshot with default props", () => {
      const { container } = render(
        <IconButton
          iconName="24/Menu"
          isProcessing={false}
        />
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    test("should match snapshot when processing", () => {
      const { container } = render(
        <IconButton
          iconName="24/Menu"
          isProcessing={true}
        />
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    test("should match snapshot with label", () => {
      const { container } = render(
        <IconButton
          iconName="24/Menu"
          isProcessing={false}
          label="Click Me"
        />
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    test("should match snapshot when disabled", () => {
      const { container } = render(
        <IconButton
          iconName="24/Menu"
          isProcessing={false}
          disabled
        />
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    test("should match snapshot with custom className", () => {
      const { container } = render(
        <IconButton
          iconName="24/Settings"
          isProcessing={false}
          className="custom-class"
        />
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    test("should match snapshot with processing and label", () => {
      const { container } = render(
        <IconButton
          iconName="24/Send"
          isProcessing={true}
          label="Sending"
        />
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe("TextButton", () => {
    test("should match snapshot with text", () => {
      const { container } = render(
        <TextButton>Click Me</TextButton>
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    test("should match snapshot when disabled", () => {
      const { container } = render(
        <TextButton disabled>Disabled Button</TextButton>
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    test("should match snapshot with custom className", () => {
      const { container } = render(
        <TextButton className="my-custom-class">
          Custom Button
        </TextButton>
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    test("should match snapshot with onClick handler", () => {
      const handleClick = jest.fn();
      const { container } = render(
        <TextButton onClick={handleClick}>
          Submit
        </TextButton>
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    test("should match snapshot with type submit", () => {
      const { container } = render(
        <TextButton type="submit">
          Submit Form
        </TextButton>
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    test("should match snapshot with long text", () => {
      const { container } = render(
        <TextButton>
          This is a very long button text that might wrap
        </TextButton>
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
