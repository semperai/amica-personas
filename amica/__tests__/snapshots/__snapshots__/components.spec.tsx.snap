// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Components > IconButton > should match snapshot when disabled 1`] = `
<button
  class="bg-primary hover:bg-primary-hover active:bg-primary-press disabled:bg-primary-disabled text-white rounded-lg text-sm p-1 text-center inline-flex items-center mr-2
        undefined
      "
  disabled=""
>
  <pixiv-icon
    name="24/Menu"
    scale="1"
  />
</button>
`;

exports[`Button Components > IconButton > should match snapshot when processing 1`] = `
<button
  class="bg-primary hover:bg-primary-hover active:bg-primary-press disabled:bg-primary-disabled text-white rounded-lg text-sm p-1 text-center inline-flex items-center mr-2
        undefined
      "
>
  <pixiv-icon
    name="24/Dot"
    scale="1"
  />
</button>
`;

exports[`Button Components > IconButton > should match snapshot with custom className 1`] = `
<button
  class="bg-primary hover:bg-primary-hover active:bg-primary-press disabled:bg-primary-disabled text-white rounded-lg text-sm p-1 text-center inline-flex items-center mr-2
        custom-class
      "
>
  <pixiv-icon
    name="24/Settings"
    scale="1"
  />
</button>
`;

exports[`Button Components > IconButton > should match snapshot with default props 1`] = `
<button
  class="bg-primary hover:bg-primary-hover active:bg-primary-press disabled:bg-primary-disabled text-white rounded-lg text-sm p-1 text-center inline-flex items-center mr-2
        undefined
      "
>
  <pixiv-icon
    name="24/Menu"
    scale="1"
  />
</button>
`;

exports[`Button Components > IconButton > should match snapshot with label 1`] = `
<button
  class="bg-primary hover:bg-primary-hover active:bg-primary-press disabled:bg-primary-disabled text-white rounded-lg text-sm p-1 text-center inline-flex items-center mr-2
        undefined
      "
>
  <pixiv-icon
    name="24/Menu"
    scale="1"
  />
  <div
    class="mx-2 font-bold"
  >
    Click Me
  </div>
</button>
`;

exports[`Button Components > IconButton > should match snapshot with processing and label 1`] = `
<button
  class="bg-primary hover:bg-primary-hover active:bg-primary-press disabled:bg-primary-disabled text-white rounded-lg text-sm p-1 text-center inline-flex items-center mr-2
        undefined
      "
>
  <pixiv-icon
    name="24/Dot"
    scale="1"
  />
  <div
    class="mx-2 font-bold"
  >
    Sending
  </div>
</button>
`;

exports[`Button Components > TextButton > should match snapshot when disabled 1`] = `
<button
  class="px-3 py-1.5 text-white text-xs font-semibold bg-slate-800 hover:bg-slate-700 active:bg-slate-900 disabled:bg-slate-400 rounded transition-all cursor-pointer undefined"
  disabled=""
>
  Disabled Button
</button>
`;

exports[`Button Components > TextButton > should match snapshot with custom className 1`] = `
<button
  class="px-3 py-1.5 text-white text-xs font-semibold bg-slate-800 hover:bg-slate-700 active:bg-slate-900 disabled:bg-slate-400 rounded transition-all cursor-pointer my-custom-class"
>
  Custom Button
</button>
`;

exports[`Button Components > TextButton > should match snapshot with long text 1`] = `
<button
  class="px-3 py-1.5 text-white text-xs font-semibold bg-slate-800 hover:bg-slate-700 active:bg-slate-900 disabled:bg-slate-400 rounded transition-all cursor-pointer undefined"
>
  This is a very long button text that might wrap
</button>
`;

exports[`Button Components > TextButton > should match snapshot with onClick handler 1`] = `
<button
  class="px-3 py-1.5 text-white text-xs font-semibold bg-slate-800 hover:bg-slate-700 active:bg-slate-900 disabled:bg-slate-400 rounded transition-all cursor-pointer undefined"
>
  Submit
</button>
`;

exports[`Button Components > TextButton > should match snapshot with text 1`] = `
<button
  class="px-3 py-1.5 text-white text-xs font-semibold bg-slate-800 hover:bg-slate-700 active:bg-slate-900 disabled:bg-slate-400 rounded transition-all cursor-pointer undefined"
>
  Click Me
</button>
`;

exports[`Button Components > TextButton > should match snapshot with type submit 1`] = `
<button
  class="px-3 py-1.5 text-white text-xs font-semibold bg-slate-800 hover:bg-slate-700 active:bg-slate-900 disabled:bg-slate-400 rounded transition-all cursor-pointer undefined"
  type="submit"
>
  Submit Form
</button>
`;

exports[`Button Components IconButton should match snapshot when disabled 1`] = `
<button
  class="bg-primary hover:bg-primary-hover active:bg-primary-press disabled:bg-primary-disabled text-white rounded-lg text-sm p-1 text-center inline-flex items-center mr-2
        undefined
      "
  disabled=""
>
  <pixiv-icon
    name="24/Menu"
    scale="1"
  />
</button>
`;

exports[`Button Components IconButton should match snapshot when processing 1`] = `
<button
  class="bg-primary hover:bg-primary-hover active:bg-primary-press disabled:bg-primary-disabled text-white rounded-lg text-sm p-1 text-center inline-flex items-center mr-2
        undefined
      "
>
  <pixiv-icon
    name="24/Dot"
    scale="1"
  />
</button>
`;

exports[`Button Components IconButton should match snapshot with custom className 1`] = `
<button
  class="bg-primary hover:bg-primary-hover active:bg-primary-press disabled:bg-primary-disabled text-white rounded-lg text-sm p-1 text-center inline-flex items-center mr-2
        custom-class
      "
>
  <pixiv-icon
    name="24/Settings"
    scale="1"
  />
</button>
`;

exports[`Button Components IconButton should match snapshot with default props 1`] = `
<button
  class="bg-primary hover:bg-primary-hover active:bg-primary-press disabled:bg-primary-disabled text-white rounded-lg text-sm p-1 text-center inline-flex items-center mr-2
        undefined
      "
>
  <pixiv-icon
    name="24/Menu"
    scale="1"
  />
</button>
`;

exports[`Button Components IconButton should match snapshot with label 1`] = `
<button
  class="bg-primary hover:bg-primary-hover active:bg-primary-press disabled:bg-primary-disabled text-white rounded-lg text-sm p-1 text-center inline-flex items-center mr-2
        undefined
      "
>
  <pixiv-icon
    name="24/Menu"
    scale="1"
  />
  <div
    class="mx-2 font-bold"
  >
    Click Me
  </div>
</button>
`;

exports[`Button Components IconButton should match snapshot with processing and label 1`] = `
<button
  class="bg-primary hover:bg-primary-hover active:bg-primary-press disabled:bg-primary-disabled text-white rounded-lg text-sm p-1 text-center inline-flex items-center mr-2
        undefined
      "
>
  <pixiv-icon
    name="24/Dot"
    scale="1"
  />
  <div
    class="mx-2 font-bold"
  >
    Sending
  </div>
</button>
`;

exports[`Button Components TextButton should match snapshot when disabled 1`] = `
<button
  class="px-3 py-1.5 text-white text-xs font-semibold bg-slate-800 hover:bg-slate-700 active:bg-slate-900 disabled:bg-slate-400 rounded transition-all cursor-pointer undefined"
  disabled=""
>
  Disabled Button
</button>
`;

exports[`Button Components TextButton should match snapshot with custom className 1`] = `
<button
  class="px-3 py-1.5 text-white text-xs font-semibold bg-slate-800 hover:bg-slate-700 active:bg-slate-900 disabled:bg-slate-400 rounded transition-all cursor-pointer my-custom-class"
>
  Custom Button
</button>
`;

exports[`Button Components TextButton should match snapshot with long text 1`] = `
<button
  class="px-3 py-1.5 text-white text-xs font-semibold bg-slate-800 hover:bg-slate-700 active:bg-slate-900 disabled:bg-slate-400 rounded transition-all cursor-pointer undefined"
>
  This is a very long button text that might wrap
</button>
`;

exports[`Button Components TextButton should match snapshot with onClick handler 1`] = `
<button
  class="px-3 py-1.5 text-white text-xs font-semibold bg-slate-800 hover:bg-slate-700 active:bg-slate-900 disabled:bg-slate-400 rounded transition-all cursor-pointer undefined"
>
  Submit
</button>
`;

exports[`Button Components TextButton should match snapshot with text 1`] = `
<button
  class="px-3 py-1.5 text-white text-xs font-semibold bg-slate-800 hover:bg-slate-700 active:bg-slate-900 disabled:bg-slate-400 rounded transition-all cursor-pointer undefined"
>
  Click Me
</button>
`;

exports[`Button Components TextButton should match snapshot with type submit 1`] = `
<button
  class="px-3 py-1.5 text-white text-xs font-semibold bg-slate-800 hover:bg-slate-700 active:bg-slate-900 disabled:bg-slate-400 rounded transition-all cursor-pointer undefined"
  type="submit"
>
  Submit Form
</button>
`;
